src/ol/Overlay.js

```diff
@@ -249,14 +249,14 @@ class Overlay extends BaseObject {
 
   /**
    * Get the map associated with this overlay.
-   * @return {import("./PluggableMap.js").default|undefined} The map that the
+   * @return {import("./PluggableMap.js").default|null} The map that the
    * overlay is part of.
    * @observable
    * @api
    */
   getMap() {
-    return /** @type {import("./PluggableMap.js").default|undefined} */ (
-      this.get(Property.MAP)
+    return /** @type {import("./PluggableMap.js").default|null} */ (
+      this.get(Property.MAP) || null
```

Comment:
Although this looks like a breaking change, it actually only fixes the inconsistency that when previously`Map#removeOverlay()` was called, a subsequent call of `Overlay#getMap()` returned `null`, but when called after `Overlay` creation, it returned `undefined`.